src/Core/Services/MetadataProviders/SqlMetadataProvider.cs

```diff
@@ -73,14 +78,15 @@ public abstract class SqlMetadataProvider<ConnectionT, DataAdapterT, CommandT> :
 
         public SqlMetadataProvider(
             RuntimeConfigProvider runtimeConfigProvider,
-            IQueryExecutor queryExecutor,
-            IQueryBuilder queryBuilder,
-            ILogger<ISqlMetadataProvider> logger)
+            IAbstractQueryManagerFactory engineFactory,
+            ILogger<ISqlMetadataProvider> logger,
+            string dataSourceName)
         {
             RuntimeConfig runtimeConfig = runtimeConfigProvider.GetConfig();
             _runtimeConfigProvider = runtimeConfigProvider;
-            _databaseType = runtimeConfig.DataSource.DatabaseType;
-            _entities = runtimeConfig.Entities;
+            _dataSourceName = dataSourceName;
+            _databaseType = runtimeConfig.GetDataSourceFromDataSourceName(dataSourceName).DatabaseType;
+            _entities = runtimeConfig.Entities.Where(x => string.Equals(runtimeConfig.GetDataSourceNameFromEntityName(x.Key), _dataSourceName, StringComparison.OrdinalIgnoreCase)).ToDictionary(x => x.Key, x => x.Value);
```

Comment:
What if there are no entities in this datasource, will the ToDictionary call succeed?